task/buildah/0.1/buildah.yaml

```diff
@@ -107,29 +107,36 @@ spec:
       sed -i 's/^short-name-mode = .*/short-name-mode = "disabled"/' /etc/containers/registries.conf
 
       if [ "${HERMETIC}" == "true" ]; then
-        NETWORK="--network none"
+        BUILDAH_ARGS="--pull=never"
+        UNSHARE_ARGS="--net"
+        for image in $(grep -i '^FROM' "$dockerfile_path" | awk '{print $2}'); do
+          buildah pull $image
+        done
         echo "Build will be executed with network isolation"
       fi
 
-      if [ ! -z "${PREFETCH_INPUT}" ]; then
+      if [ -n "${PREFETCH_INPUT}" ]; then
         VOLUME_MOUNTS="
           --volume $(realpath ./cachi2/output):/cachi2/output:U \
           --volume $(realpath ./cachi2/cachi2.env):/cachi2/cachi2.env:U"
-
+        sed -i 's|^\s*run |RUN source /cachi2/cachi2.env \&\& \\\n    |i' "$dockerfile_path"
```

Comment:
Maybe the leading whitespace in `^\s*run` doesn't need to be there

Technically buildah will consider this Dockerfile valid, but the rest of the code in this task does not accept leading whitespace

```dockerfile
FROM alpine:latest

    RUN echo hello there
```

Alternatively, other commands in this task would have have to be updated to accept leading whitespace

Reply:
added